File: app/src/main/java/com/intkhabahmed/smartnotes/utils/NotesReminderTasks.java
package com.intkhabahmed.smartnotes.utils;

import android.content.Context;

import com.intkhabahmed.smartnotes.models.Note;

public class NotesReminderTasks {

    public static void executeTask(Context context, String action, Note note) {
        switch (action) {
            case AppConstants.NOTE_REMINDER_TASK:
                startNoteReminderTask(context, note);
                break;
            default:
                throw new IllegalArgumentException();
        }
    }

    private static void startNoteReminderTask(final Context context, Note note) {
        if (note != null) {
            NotificationUtils.showReminderNotification(context, note);
        }
    }
}

File: app/src/main/java/com/intkhabahmed/smartnotes/utils/NotesReminderTasks.kt
package com.intkhabahmed.smartnotes.utils

import android.content.Context
import com.intkhabahmed.smartnotes.models.Note
import com.intkhabahmed.smartnotes.utils.NotificationUtils.showReminderNotification

object NotesReminderTasks {
    @JvmStatic
    fun executeTask(context: Context, action: String, note: Note?) {
        if (AppConstants.NOTE_REMINDER_TASK == action) {
            startNoteReminderTask(context, note)
        } else {
            throw IllegalArgumentException()
        }
    }

    private fun startNoteReminderTask(context: Context, note: Note?) {
        note?.let { showReminderNotification(context, it) }
    }
}